mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala

```diff
@@ -328,74 +336,69 @@ class IsotonicRegression private (private var isotonic: Boolean) extends Seriali
       return Array.empty
     }
 
-    // Pools sub array within given bounds assigning weighted average value to all elements.
-    def pool(input: Array[(Double, Double, Double)], start: Int, end: Int): Unit = {
-      val poolSubArray = input.slice(start, end + 1)
 
-      val weightedSum = poolSubArray.map(lp => lp._1 * lp._3).sum
-      val weight = poolSubArray.map(_._3).sum
+    // Keeps track of the start and end indices of the blocks. blockBounds(start) gives the
+    // index of the end of the block and blockBounds(end) gives the index of the start of the
+    // block. Entries that are not the start or end of the block are meaningless. The idea is that
```

Comment:
I'm still not sure about this comment -- how can `blockBounds(x)` be both the start and end of a block? the implementation below is identical.